### ISSUE TITLE
Add prefix filter search in select control

### ISSUE THREAD

**Issue by gbataille** [tmbo/questionary#33]
As featured here (when selecting the AWS region)
https://asciinema.org/a/AyujEPdjcDmSPoOK26pTozCiO

@tmbo thanks for pointing me here from PyInquirer. Looks like the more modern/maintained alternatives (latest lib, tests, ...)

I'll try to use questionary soon and if it's all good, I'll contribute the filter that I have built that I find very practical on long lists like the one I'm dealing with.

**Comment by tmbo** [comment]
The filter looks really cool, would love to integrate that :rocket:

**Comment by gbataille** [comment]
Hey @tmbo 
so I migrated my code to `questionary`. https://github.com/gbataille/password-organizer/pull/1

What I like
- you created a `Choice` class (like I had done) 👍 
- I could remove all my custom code (at least the part that was emulating PyInquirer)

What I'm missing
- I think `Choice` should be a `Generic` type, parameterised by the type `T` of it's return value. Actually I would personally tend to remove the python-style "I accept multiple type in input" and just deal with `List` of `Choice` and not accept `str` and `dict`
- You swallow `KeyboardInterrupt` errors to return `None`. I had made the choice to not do that. Not sure I like it, but at least we should document it.
- The current implementation does not allow custom key bindings. That's something I'll have to work on for the **prefix search** feature and see how to not make it collide with the shortcut feature and the vim-based navigation in the menu (maybe one of the other, or a chord to launch the search)
- It would be nice to but the **Back** feature that I have (https://github.com/gbataille/password-organizer/blob/master/password_organizer/menu.py#L49) in the lib
- I'm missing the highlighting of the currently selected answer, but I guess that's just a default config/style I have to change somewhere
...

I'll try and do individual issues/PR in your repo for all those points, unless you feel otherwise

Cheers
